pkg/tsdb/prometheus/querydata/exemplar/framer.go

```diff
@@ -64,8 +64,14 @@ func (f *Framer) Frames() (data.Frames, error) {
 		timeField.Append(b.Timestamp)
 		valueField.Append(b.Value)
 		for i, labelName := range labelNames {
```

Comment:
would it make sense to move the label fetching logic to the label tracker? 

something like:

```go
for i, labelValue := f.labelTracker.GetLabels(ex) {
  colIdx := i + 2 // +2 to skip time and value fields
  exemplarFrame.Fields[colIdx].Append(labelValue)
}
```

Reply:
I think one should at least pass `labelNames` to `f.labelTracker.GetLabels`, to avoid the performance hit of having to recompute it for every iteration. With that complication in mind, is it even worth factoring out the code?

I think such a solution should only contribute negatively to performance, right, since `f.labelTracker.GetLabels` would have to make an intermediate list (of label values).